azure/scope/managedcontrolplane.go

```diff
@@ -300,3 +313,171 @@ func (s *ManagedControlPlaneScope) GetPrivateDNSZoneName() string {
+	}
+
+	managedClusterSpec := azure.ManagedClusterSpec{
+		Name:                  s.ControlPlane.Name,
+		ResourceGroupName:     s.ControlPlane.Spec.ResourceGroupName,
+		NodeResourceGroupName: s.ControlPlane.Spec.NodeResourceGroupName,
+		Location:              s.ControlPlane.Spec.Location,
+		Tags:                  s.ControlPlane.Spec.AdditionalTags,
+		Version:               strings.TrimPrefix(s.ControlPlane.Spec.Version, "v"),
+		SSHPublicKey:          string(decodedSSHPublicKey),
+		DNSServiceIP:          s.ControlPlane.Spec.DNSServiceIP,
+		VnetSubnetID: azure.SubnetID(
+			s.ControlPlane.Spec.SubscriptionID,
+			s.ControlPlane.Spec.ResourceGroupName,
+			s.ControlPlane.Spec.VirtualNetwork.Name,
+			s.ControlPlane.Spec.VirtualNetwork.Subnet.Name,
+		),
+	}
+
+	if s.ControlPlane.Spec.NetworkPlugin != nil {
+		managedClusterSpec.NetworkPlugin = *s.ControlPlane.Spec.NetworkPlugin
+	}
+	if s.ControlPlane.Spec.NetworkPolicy != nil {
+		managedClusterSpec.NetworkPolicy = *s.ControlPlane.Spec.NetworkPolicy
+	}
+	if s.ControlPlane.Spec.LoadBalancerSKU != nil {
+		managedClusterSpec.LoadBalancerSKU = *s.ControlPlane.Spec.LoadBalancerSKU
+	}
+
+	if net := s.Cluster.Spec.ClusterNetwork; net != nil {
+		if net.Services != nil {
+			// A user may provide zero or one CIDR blocks. If they provide an empty array,
+			// we ignore it and use the default. AKS doesn't support > 1 Service/Pod CIDR.
+			if len(net.Services.CIDRBlocks) > 1 {
+				return azure.ManagedClusterSpec{}, errors.New("managed control planes only allow one service cidr")
+			}
+			if len(net.Services.CIDRBlocks) == 1 {
+				managedClusterSpec.ServiceCIDR = net.Services.CIDRBlocks[0]
+			}
+		}
+		if net.Pods != nil {
+			// A user may provide zero or one CIDR blocks. If they provide an empty array,
+			// we ignore it and use the default. AKS doesn't support > 1 Service/Pod CIDR.
+			if len(net.Pods.CIDRBlocks) > 1 {
+				return azure.ManagedClusterSpec{}, errors.New("managed control planes only allow one service cidr")
+			}
+			if len(net.Pods.CIDRBlocks) == 1 {
+				managedClusterSpec.PodCIDR = net.Pods.CIDRBlocks[0]
+			}
+		}
+	}
+
+	if s.ControlPlane.Spec.DNSServiceIP != nil {
+		if managedClusterSpec.ServiceCIDR == "" {
+			return azure.ManagedClusterSpec{}, fmt.Errorf(s.Cluster.Name + " cluster serviceCIDR must be specified if specifying DNSServiceIP")
+		}
+		_, cidr, err := net.ParseCIDR(managedClusterSpec.ServiceCIDR)
+		if err != nil {
+			return azure.ManagedClusterSpec{}, fmt.Errorf("failed to parse cluster service cidr: %w", err)
+		}
+		ip := net.ParseIP(*s.ControlPlane.Spec.DNSServiceIP)
+		if !cidr.Contains(ip) {
+			return azure.ManagedClusterSpec{}, fmt.Errorf(s.ControlPlane.Name + " DNSServiceIP must reside within the associated cluster serviceCIDR")
+		}
+	}
+
+	return managedClusterSpec, nil
+}
+
+// GetDefaultAgentPoolSpec gets azure.AgentPoolSpec for default agent pool.
+func (s *ManagedControlPlaneScope) GetDefaultAgentPoolSpec() azure.AgentPoolSpec {
```

Comment:
Are we using this function anywere?

Reply:
Good catch, I added this one first, then replaced it with the one that returns system node pools. I'll remove it.

Reply:
Removed this unused function.